test/e2e/velero_utils.go

```diff
@@ -395,3 +420,51 @@ func VeleroAddPluginsForProvider(ctx context.Context, veleroCLI string, veleroNa
 
 	return nil
 }
+
+/*
+ Waits for uploads started by the Velero Plug-in for vSphere to complete
+ TODO - remove after upload progress monitoring is implemented
+*/
+func waitForVSphereUploadCompletion(ctx context.Context, timeout time.Duration, namespace string) error {
+	err := wait.PollImmediate(time.Minute, timeout, func() (bool, error) {
+		checkSnapshotCmd := exec.CommandContext(ctx, "kubectl",
+			"get", "-n", namespace, "snapshots.backupdriver.cnsdp.vmware.com", "-o=jsonpath='{range .items[*]}{.spec.resourceHandle.name}{\"=\"}{.status.phase}{\"\\n\"}'")
+		fmt.Printf("checkSnapshotCmd cmd =%v\n", checkSnapshotCmd)
+		stdout, stderr, err := veleroexec.RunCommand(checkSnapshotCmd)
+		if err != nil {
+			fmt.Print(stdout)
+			fmt.Print(stderr)
+			return false, errors.Wrap(err, "failed to verify")
+		}
+		lines := strings.Split(stdout, "\n")
+		complete := true
+		for _, curLine := range lines {
+			fmt.Println(curLine)
+			comps := strings.Split(curLine, "=")
+			// SnapshotPhase represents the lifecycle phase of a Snapshot.
+			// New - No work yet, next phase is InProgress
+			// InProgress - snapshot being taken
+			// Snapshotted - local snapshot complete, next phase is Protecting or SnapshotFailed
+			// SnapshotFailed - end state, snapshot was not able to be taken
+			// Uploading - snapshot is being moved to durable storage
+			// Uploaded - end state, snapshot has been protected
+			// UploadFailed - end state, unable to move to durable storage
+			// Canceling - when the SanpshotCancel flag is set, if the Snapshot has not already moved into a terminal state, the
+			//             status will move to Canceling.  The snapshot ID will be removed from the status status if has been filled in
+			//             and the snapshot ID will not longer be valid for a Clone operation
+			// Canceled - the operation was canceled, the snapshot ID is not valid
+			if len(comps) == 2 {
+				phase := comps[1]
+				if phase == "New" ||
+					phase == "InProgress" ||
+					phase == "Snapshotted" ||
+					phase == "Uploading" {
+					complete = false
+				}
+			}
+		}
+		return complete, nil
```

Comment:
Do we want to return an error in the `SnapshotFailed`, `UploadFailed` or cancel phases? With the current use, the test will continue even if we fail to upload. That may be okay, but I just want to make sure that we still have visibility of this failure when the restore step in the test fails later.

Reply:
The status will get logged currently, we output that up above.  I didn't error out here because it's possible that the there are records floating around from previous backups and I didn't want us to start being flakey because stuff was hanging around.